internal/enginecache/cache.go

```diff
@@ -284,68 +234,67 @@ func ValidateOrUpdateEngine(targetOS, cachePath, requiredEngineVersion string) {
 
 	engineZipPath := filepath.Join(dir, "engine.zip")
 	engineExtractPath := filepath.Join(dir, "engine")
-	artifactsZipPath := filepath.Join(dir, "artifacts.zip")
 
-	log.Printf("Downloading engine for platform %s at version %s...", platform, requiredEngineVersion)
-	err = downloadFile(engineZipPath, engineDownloadURL)
-	if err != nil {
-		log.Errorf("Failed to download engine: %v", err)
-		os.Exit(1)
-	}
+	log.Printf("Downloading engine for platform %s at version %s...", EngineConfig(targetOS, mode), requiredEngineVersion)
 
-	// TODO, optimization: make artifacts download a separate function, it doesn't need to be
-	// downloaded with engine because it's OS independent.
-	log.Printf("Downloading artifacts at version %s...", requiredEngineVersion)
-	err = downloadFile(artifactsZipPath, icudtlDownloadURL)
-	if err != nil {
-		log.Errorf("Failed to download artifacts: %v", err)
-		os.Exit(1)
+	file := ""
+	switch targetOS {
+	case "linux":
+		file += "linux"
+	case "darwin":
+		file += "macosx"
+	case "windows":
+		file += "windows"
 	}
+	file += fmt.Sprintf("_x64-host_%s.zip", mode.Name)
+	engineDownloadURL := fmt.Sprintf("https://github.com/flutter-rs/engine-builds/releases/download/f-%s/%s", requiredEngineVersion, file)
 
-	_, err = unzip(engineZipPath, engineExtractPath) // engineCachePath)
+	err = downloadFile(engineZipPath, engineDownloadURL)
 	if err != nil {
-		log.Warnf("%v", err)
+		log.Errorf("Failed to download engine: %v", err)
+		log.Infof("That may mean no engine download is currently available. You'll have to wait for one to get available")
```

Comment:
Maybe we can improve the wording a bit here and give better advice.

A possible "solution" for the user may be to go back an older flutter version. We could advice the user to use the beta channel if they aren't using that already, since those engines are more likely to be available.

We may also explain where we get the engine builds from, so the user knows where to look which versions are available.

Reply:
Looking up the versions manually isn't realy fun. We could just download the engine versions for all branches and then tell the user which ones are already built so they can use that branch.

Reply:
I've changed the wording, but it's still not perfect